corehq/apps/users/tasks.py

```diff
@@ -303,3 +306,33 @@ def update_domain_date(user_id, domain):
             and today > domain_membership.last_accessed):
         domain_membership.last_accessed = today
         user.save()
+
+
+@periodic_task(
+    run_every=deserialize_run_every_setting(settings.USER_REPORTING_METADATA_BATCH_SCHEDULE),
+    queue='background_queue',
+)
+def process_reporting_metadata_staging():
+    from corehq.apps.users.models import UserReportingMetadataStaging
+    from corehq.pillows.synclog import mark_last_synclog
+    from pillowtop.processors.form import mark_latest_submission
+
+    records = (
+        UserReportingMetadataStaging.objects.select_for_update(skip_locked=True).order_by('pk')
```

Comment:
This is not a pattern we use currently. The django docs exist [here](https://docs.djangoproject.com/en/1.11/ref/models/querysets/#django.db.models.query.QuerySet.select_for_update)

The idea here is that I'm going to grab the first 100 rows that are not being used in a current query. This query locks (in postgres) the rows that are being evaluated when we enter the transaction. (It is outside the transaction because querysets in django are lazy).

The reason to use `skip_locked` is so that multiple of these tasks can run safely at one time without clobbering each other and also they shouldn't lock on one another.

This lock should also prevent the record from being retrieved in the pillows. I do not expect the locking to be an issue for pillows because we will be grabbing only 100 rows at one time. We can lower this if we want to make this task even smaller